src/android/CodePush.java

```diff
@@ -103,13 +103,24 @@ protected Void doInBackground(Void... params) {
         return true;
     }
 
-    private boolean execUpdateSuccess(CallbackContext callbackContext) {
+    private boolean execUpdateSuccess(CordovaArgs args, CallbackContext callbackContext) {
+        String deploymentKey = "";
+        try {
+            deploymentKey = args.getString(0);
+        } catch (JSONException e) {
+            // no-op becuase we don't care if we didn't get a deployment key from script; we will just try to use the one from preferences
+        }
+
+        if (deploymentKey.equals(null) || deploymentKey.equals("") || deploymentKey.equals("null") || deploymentKey.equals("undefined")) {
```

Comment:
`deploymentKey.equals(null)` will never be true. If `deploymentKey == null`,  `deploymentKey.equals(null)` will throw a `NullPointerException`.